### ISSUE TITLE
[Enhancement]: Add Filter [All / Audiobooks Only / E-Books Only]

### ISSUE THREAD

**Issue by pekempy** [advplyr/audiobookshelf#3255]
### Type of Enhancement

Web Interface/Frontend

### Describe the Feature/Enhancement

A filter which allows you to filter only to audiobooks, e-books or all results

### Why would this be helpful?

If users have their library stored together to achieve the `Play` / `Read` buttons adjacent to each other, some books may be missing the audiobooks, or some audiobooks may not have an e-book counterpart.
This filter would allow people to only display entries in the Library which have an audiobook, if they want to listen to one, or just to ebooks.

The 'Tracks' sort of works this way already, but I can't display both Multi-track and single-track together - with larger libraries it can become a problem.

### Future Implementation (Screenshot)

![image](https://github.com/user-attachments/assets/010ae924-8d05-403f-a572-0d8b0c13c7cc)
(or just as a submenu in the existing filter as 'Media Type' -> 'All' / 'Audiobooks' / 'E-Books'

### Audiobookshelf Server Version

2.12.1

### Current Implementation (Screenshot)

![image](https://github.com/user-attachments/assets/f264d131-856c-41b5-b3cd-397110a0dd71)

**Comment by nichwall** [comment]
Related to https://github.com/advplyr/audiobookshelf/issues/1465

You can already filter by presence of ebooks

![Screenshot_20240807-134451.png](https://github.com/user-attachments/assets/2eb80c9f-060b-4189-99af-32c071a07550)

![Screenshot_20240807-134617.png](https://github.com/user-attachments/assets/0096475c-80ae-42b5-9b12-1e3782ec538e)

**Comment by pekempy** [comment]
Then maybe just an option for `Has tracks` in the tracks submenu, which would show any audiobooks and exclude ebook only?

**Comment by advplyr** [comment]
"Has no ebook" would mean that it has tracks otherwise it wouldn't be able to get scanned in.

To select all books that are ebook only you would select "Tracks->No tracks"
To select all books that are audiobook only you would select "Ebooks->Has no ebook"

**Comment by pekempy** [comment]
I think I might be being misunderstood, so renamed it
**Audiobooks** would show:
- Audiobooks only
- Audiobooks & Ebooks

**Ebooks** would show:
- Ebooks only
- Audiobooks & Ebooks

**All** would be the default / cleared filter, and would show all library items.

Currently there's no way to filter the library to only show _any_ item which contains an audiobook that I've found (Regardless of multi/single track, I can filter to single track OR multi-track, but not both at once) - Glad to be corrected if I am wrong

**Comment by advplyr** [comment]
I see. I think adding "Has Tracks" to the "Tracks" menu would be the best solution

**Comment by sydlexius** [comment]
This seems unintuitive, since these are two different categories to perform similar functions.  My preference would be to have a media types category, and nest most if not all of the options in Tracks and Ebooks under that instead.

**Comment by advplyr** [comment]
Media types makes sense. If we add that we should remove both Tracks and Ebooks. What would the options for Media types be so that it covers all use-cases?

**Comment by pekempy** [comment]
Media Types > Audiobooks > Has Tracks
Media Types > Audiobooks > Single track
Media Types > Audiobooks > Multi track
Media Types > Audiobooks > No Tracks
Media Types > Ebooks > Has Ebooks
Media Types > Ebooks > Has Supplementary Ebooks
Media Types > Ebooks > No Ebooks

Then obviously it would be clearable to show everything.

**Comment by advplyr** [comment]
Probably want to avoid double nesting like that if we can. The component would have to be written to support that and it is probably not necessary

**Comment by sydlexius** [comment]
This is where I wish ABS supported multiple library filters (such as req #1465), because we're dealing with at least 4 bits of information (single track, multi track, ebook, supplementary ebooks), that leaves a configuration matrix of 16 possible combinations for just those 4 types of option.  To enumerate each possible combination isn't going to be pretty.

With that said, at the bare minimum one may want to know if a title contains an audiobook, ebook, or both (ignoring single vs. multitrack, regular vs. supplementary ebooks).

**Comment by advplyr** [comment]
Yeah I'd like to do multi-filter but it will probably be a while before that happens if it does. Since we are still half-way through migrating to the new data model the sql queries as they are now are barely manageable.

We won't cover every possible permutation but we can cover the ones existing and the new one discussed in this thread.

**Comment by adapowers** [comment]
Just adding a +1 to this issue. ABS is, unfortunately, a victim of its amazing work with audiobooks, and now my entire workflow is built around it, including ebooks. Quick access to an ebook/audiobook toggle would be amazing for my users who go the app to listen, but send books to their Kindles to read.

**Comment by jean874** [comment]
+1 to this. I am currently using a separate book library only for eBooks because there is no easy way to filter titles that have only an eBook or only an audiobook. I have some cases where I only own the eBook, and it creates a poor user experience to constantly guess whether an audiobook is attached to it or not. A simple top-level filter (Audiobook / Both / eBook) would completely solve this issue and allow me to use a single library for all my titles, which would be much more convenient.
